src/common/src/com/microsoftopentechnologies/tooling/msservices/helpers/auth/UserInfo.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright 2014 Microsoft Open Technologies Inc.
+ * <p/>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.microsoftopentechnologies.tooling.msservices.helpers.auth;
+
+import com.microsoftopentechnologies.tooling.msservices.helpers.NotNull;
+
+public class UserInfo {
+    private final String tenantId;
+    private final String uniqueName;
+
+    public UserInfo(@NotNull String tenantId, @NotNull String uniqueName) {
+        this.tenantId = tenantId;
+        this.uniqueName = uniqueName;
+    }
+
+    @NotNull
+    public String getTenantId() {
+        return this.tenantId;
+    }
+
+    @NotNull
+    public String getUniqueName() {
+        return this.uniqueName;
+    }
+
+    @Override
+    public boolean equals(Object otherObj) {
+        if (otherObj != null && otherObj instanceof UserInfo) {
+            UserInfo other = (UserInfo) otherObj;
+            return tenantId.equals(other.tenantId) && uniqueName.equals(other.uniqueName);
+        }
+
+        return false;
+    }
+
+    @Override
+    public int hashCode() {
+        return tenantId.hashCode() * 13 + uniqueName.hashCode();
```

Comment:
I'd rather avoid calculating hashes based on string concatenation, to prevent things like "abc" + "de" =  "ab" + "cde"
The current implementation is quite standard, in fact is based on the proposed hashcode implementation for lists https://docs.oracle.com/javase/6/docs/api/java/util/List.html#hashCode()